attest/core/src/types/report_body.rs

```diff
@@ -173,9 +173,13 @@ impl ReportBody {
         // Check mr_signer/mr_enclave
         let mr_signer = self.mr_signer();
         let mr_enclave = self.mr_enclave();
-        if expected_measurement != &mr_signer && expected_measurement != &mr_enclave {
+        let matching_measurements: Vec<&Measurement> = expected_measurements
```

Comment:
I was a little surprised by the "or" in the filter condition (I would have guessed "and"). Might be worth changing the comment "Check mr_signer/mr_enclave" to describe what condition is being checked. 

This also might be a good place for iter().find(...) instead of iter().filter(...), but what you've got here looks good.